torch/distributed/distributed_c10d.py

```diff
@@ -1451,6 +1455,7 @@ def destroy_process_group(group: Optional[ProcessGroup] = None):
         _world.tags_to_pg.clear()
         _world.pg_coalesce_state.clear()
         _world.pg_default_device.clear()
+        _unregister_all_process_groups()
```

Comment:
So IIUC, once we have more verification and testing, we can remove the whole python layer global map and universally use C++ global map, and then we can get rid of the double state maintainace across our codebase, is this right?

Reply:
Yes exactly